fiona/ogrext.pyx

```diff
@@ -760,26 +760,30 @@ cdef class OGRFeatureBuilder:
                 OGR_F_SetFieldNull(cogr_feature, i)
             else:
                 schema_type = session._schema_normalized_field_types[key]
-                fieldkey = (*FIELD_TYPES_MAP2[NAMED_FIELD_TYPES[schema_type]], type(value).__name__)
-                if fieldkey in self.property_setter_cache:
-                    setter = self.property_setter_cache[fieldkey]
+                val_type = type(value)
+
+                if val_type in self.property_setter_cache:
+                    setter = self.property_setter_cache[val_type]
```

Comment:
In a nutshell: the code in 1.9 tested `elif isinstance(value, (date, time, datetime)` before calling a field setter. Now we register setters to names like "date" and use the MRO of `type(value)` to look up and cache a setter.

Reply:
Look okay to you @mwtoews ?